File: Python/gloma.py
"""
|--------------------------------------------------------------------------
| Length of missing array
|--------------------------------------------------------------------------
|
"""

def get_length_of_missing_array(x):
   if x == None or len(x) == 0:
      return 0
   else:
      res = []
      for i in range(len(x)):
         if x[i] == None:
            return 0
         else:
            res.append(len(x[i]))
      arr = sorted(res)
      if arr[0] == 0:
         return 0
      else:
         last = arr[len(arr)-1]
         i = 0
         for j in range(arr[0], last):
            if j != arr[i]:
               return j
            i+=1


def get_length_of_missing_array(a):
   lns = a and all(a) and list(map(len, a))
   return bool(lns) and sum(range(min(lns), max(lns) + 1)) - sum(lns)


def get_length_of_missing_array(arr):
   if not arr or any(type(a) != list or not len(a) for a in arr):
      return 0
   lengths = set(len(a) for a in arr)
   return (set(range(min(lengths), max(lengths)+1)) - set(lengths)).pop()


def get_length_of_missing_array(array_of_arrays):
   try:
      lengths = [len(array) for array in array_of_arrays]
      return 0 if 0 in lengths \
         else [missing for missing in range(min(lengths), max(lengths)) if missing not in lengths][0]
   except:
      return 0


print(get_length_of_missing_array([
   [1, 0, 3, 3, 3, 1, 2, 3, 4, 0, 4],
   [4, 2],
   [1, 4, 4, 3, 1],
   [3, 1, 2, 0, 2, 4, 1],
   [],
   [3, 1, 0],
   [1, 1, 2, 0],
   [2, 4, 0, 3, 2, 2, 3, 0],
   [4, 0, 1, 0, 0, 4],
   [2, 4, 3, 0, 0, 3, 3, 0, 3],
   [4, 4, 3, 3, 2, 3, 4, 0, 0, 2]
]), 0)
print(get_length_of_missing_array([]), 0)
print(get_length_of_missing_array([None]), 0)

File: PHP/gloma.php
<?php
/*
|--------------------------------------------------------------------------
| Length of missing array
|--------------------------------------------------------------------------
|
*/

function getLengthOfMissingArray($x) {
   if ($x == null || count($x) == 0) {
      return 0;
   } else {
      $res = [];
      for ($i = 0; $i < count($x); $i++) {
         if ($x[$i] == null) {
            return 0;
         } else {
            array_push($res, count($x[$i]));
         }
      }
      sort($res);
      if ($res[0] == 0) {
         return 0;
      } else {
         $last = $res[count($res) - 1];
         $i = 0;
         for ($j = $res[0]; $j <= $last; $j++) {
            if ($j != $res[$i]) {
               return $j;
            }
            $i += 1;
         }
      }
   }
}


function getLengthOfMissingArray($arrayOfArrays) {
   if(!$arrayOfArrays || in_array(0, $length_array = array_map('count', $arrayOfArrays))) return 0;
   $compare_array = range(min($length_array),max($length_array));
   return array_values(array_diff($compare_array, $length_array))[0];
}


function getLengthOfMissingArray($arrayOfArrays) {
   if (empty( $arrayOfArrays ) || in_array(0, $counts = array_map('count', $arrayOfArrays) ) ) return 0;
   return min(array_diff(range(min($counts),max($counts)), $counts));
}


function getLengthOfMissingArray($arrayOfArrays) {
   $a = array_map('count', (array)$arrayOfArrays);
   return $a && min($a) ? array_sum(range(min($a), max($a))) - array_sum($a) : 0;
}


print_r(getLengthOfMissingArray([
   [1, 0, 3, 3, 3, 1, 2, 3, 4, 0, 4],
   [4, 2],
   [1, 4, 4, 3, 1],
   [3, 1, 2, 0, 2, 4, 1],
   [],
   [3, 1, 0],
   [1, 1, 2, 0],
   [2, 4, 0, 3, 2, 2, 3, 0],
   [4, 0, 1, 0, 0, 4],
   [2, 4, 3, 0, 0, 3, 3, 0, 3],
   [4, 4, 3, 3, 2, 3, 4, 0, 0, 2]
]) . ' ' . 0 . '
');
print_r(getLengthOfMissingArray([]) . ' ' . 0 . '
');
print_r(getLengthOfMissingArray([null]) . ' ' . 0) . '
'; 